rust/lance/src/dataset.rs

```diff
@@ -1163,6 +1163,16 @@ impl Dataset {
         Ok(versions)
     }
 
+    /// Get the latest version of the dataset
+    /// This is meant to be a fast path for checking if a dataset has changed. This is why
+    /// we don't return the full version struct.
+    pub async fn latest_version_id(&self) -> Result<u64> {
```

Comment:
Shall we store version in manifest tho?  Listing over handreds of versions over S3 is slower than open a file ?

Reply:
https://github.com/lancedb/lance/blob/eadeaa70c5da961f3a6c2c4b36f2374b713afd47/protos/format.proto#L56

Reply:
the version id stored in manifest isn't the latest.

This api is really meant for dataset that use external manfiest store, which usually implement O(1) lookup on latest version

Reply:
once we migration to `_latest.json` and zero-padded manifest files, I think even listing object store will be fast enough

Reply:
> the version id stored in manifest isn't the latest.

What was it?  Is it a bug that it is not the same?